XenAdmin/Wizards/GenericPages/SelectMultipleVMDestinationPage.cs

```diff
@@ -404,16 +405,21 @@ private void PopulateDataGridView(IEnableableXenObjectComboBoxItem selectedItem)
                             }
                         }
 
-                        foreach (var host in Connection.Cache.Hosts)
+                        var sortedHosts = new List<Host>(Connection.Cache.Hosts);
+                        sortedHosts.Sort();
+
+                        var items = new List<DelayLoadingOptionComboBoxItem>();
+
+                        foreach (var host in sortedHosts)
                         {
                             var item = new DelayLoadingOptionComboBoxItem(host, homeserverFilters);
                             item.LoadAndWait();
                             cb.Items.Add(item);
-
-                            if (item.Enabled && ((m_selectedObject != null && m_selectedObject.opaque_ref == host.opaque_ref) ||
-                                (target != null && target.Item.opaque_ref == host.opaque_ref)))
+                            if (item.Enabled && ((m_selectedObject != null && m_selectedObject.opaque_ref == item.Item.opaque_ref) ||

```

Comment:
If we revert to using `host.opaque_ref` instead of `item.Item.opaque_ref`, the overall diff will show no changes here.